src/Main/Character/CharacterParses.js

```diff
@@ -0,0 +1,482 @@
+    this.fetchImage();
+    this.load();
+  }
+
+  iconPath(specName) {
+    return `/specs/${this.state.class.replace(" ", "")}-${specName.replace(" ", "")}.jpg`;
+  }
+
+  updateDifficulty(diff) {
+    let newDiff = this.state.activeDifficulty;
+    if (newDiff.includes(diff)) {
+      newDiff = newDiff.filter((elem, index) => {
+        return elem !== diff;
+      });
+    } else {
+      newDiff.push(diff);
+    }
+
+    this.setState({
+      activeDifficulty: newDiff,
+    });
+  }
+
+  updateSpec(spec) {
+    let newSpec = this.state.activeSpec;
+    if (newSpec.includes(spec)) {
+      newSpec = newSpec.filter((elem, index) => {
+        return elem !== spec;
+      });
+    } else {
+      newSpec.push(spec);
+    }
+
+    this.setState({
+      activeSpec: newSpec,
+    });
+  }
+
+  fillMissingTrinkets() {
+    Object.values(ITEMS).forEach(trinket => {
+      if (trinket.icon === ITEMS[0].icon && trinket.id !== 0) {
+        return fetch(`https://eu.api.battle.net/wow/item/${trinket.id}?locale=en_GB&apikey=n6q3eyvqh2v4gz8t893mjjgxsf9kjdgz`)
+          .then(response => response.json())
+          .then((data) => {
+            ITEMS[trinket.id].icon = data.icon;
+            this.forceUpdate();
+          });
+      }
+    });
+  }
+
+  get filterParses() {
+    let filteredParses = this.state.parses;
+    filteredParses = filteredParses
+      .filter((elem, index) => { return this.state.activeDifficulty.includes(elem.difficulty); })
+      .filter((elem, index) => { return this.state.activeSpec.includes(elem.spec); })
+      .sort((a, b) => { 
+        if (this.state.sortBy === 0) {
+          return b.start_time - a.start_time; 
+        } else if (this.state.sortBy === 1) {
+          return b.persecondamount - a.persecondamount;
+        }
+        return b.historical_percent - a.historical_percent;
+      });
+
+    if (Number(this.state.activeEncounter) === 0) {
+      return filteredParses.slice(0, RENDER_LIMIT);
+    }
+
+    filteredParses = filteredParses.filter((elem, index) => {
+      return elem.name === this.state.activeEncounter;
+    });
+
+    return filteredParses.slice(0, RENDER_LIMIT);
+  }
+
+  rankingColor(rank) {
+    rank = Math.floor(rank);
+    switch (true) {
+      case (rank < 25):
+        return 'grey';
+      case (rank < 50):
+        return 'green';
+      case (rank < 75):
+        return 'blue';
+      case (rank < 95):
+        return 'purple';
+      case (rank <= 99):
+        return 'orange';
+      case (rank === 100):
+        return 'artifact';
+      default:
+        return 'none';
+    }
+  }
+
+  //resolve the boss+difficulty->spec->parse structure to make sorting & filtering easier
+  changeParseStructure(rawParses) {
+    const parses = [];
+    rawParses.forEach((elem, index) => {
+      const name = elem.name;
+      const difficulty = DIFFICULTIES[elem.difficulty];
+
+      elem.specs
+        .filter((item) => { return item.spec !== "Melee" && item.spec !== "Ranged"; })
```

Comment:
https://www.warcraftlogs.com/rankings/character/931710/17/#metric=dps

Specs with two or more DPS specs have an extra class called "Ranged" || "Melee". Its basically a duplicated entry or the original spec, just with the spec as "Ranged". I reckon Kihra did this to make it easier to make it easier for the character-ranking page.
Either way, I think Kihra mentioned that he's thinking about removing those "cross spec" tabs as they don't have much use. And we don't need them because we allow filtering by multiple specs anyways.